### DISCUSSION TITLE
How to apply "build from source" option recursively?

### DISCUSSION THREAD

**Discussion by leavelet** [Homebrew/discussions#3367] [Everyday usage] [answered]
Hi here!
I am using macOS 13.0 developer's beta 1, with no available bottle to use. So I use the "build from source" option to install formulas. But when installing its deps, same problem appears and I have to manually install every deps for that formula, which is quite annoying.
I am wondering if there is an option to apply "build from source" recursively, which will make the process a lot easier. I've seached docs and Common Issues, but I don't get an answer.

**Comment by SMillerDev** [comment]
Bottles from older OS versions should work just fine. There might be a environment variable for source builds but it'll really make your life much harder than using the Monterey bottles.

**Comment by carlocab** [comment]
Try
```bash
brew install --build-from-source $(brew deps foo) foo
```
If that doesn't work,
```bash
for f in $(brew deps -n foo) foo
do
  brew install --build-from-source $f || break
done
```

**Reply by leavelet** [reply]
sounds good, thank you!

**Comment by Bo98** [comment, marked as answer]
> I am using macOS 13.0 developer's beta 1, with no available bottle to use.

It should be falling back to macOS 12 bottles. What does `brew config` output?

**Reply by leavelet** [reply]
```
HOMEBREW_VERSION: 3.4.9
ORIGIN: https://github.com/leavelet/brew.git
HEAD: d0a0bbef8db45cca303c607037338d02d8eaa8ca
Last commit: 7 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: adb4552a948d040e7c6ee231df8dcebdd7ccf259
Core tap last commit: 12 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BOTTLE_DOMAIN: https://mirrors.tuna.tsinghua.edu.cn/homebrew-bottles
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.9 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.36.1 => /opt/homebrew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 13.0-arm64
CLT: 14.0.0.0.1.1654301352
Xcode: 14.0 => /Applications/Xcode-beta.app/Contents/Developer
Rosetta 2: false
```

**Reply by leavelet** [reply]
So how can I solve the problem?

**Reply by Bo98** [reply]
```
HOMEBREW_VERSION: 3.4.9
ORIGIN: https://github.com/leavelet/brew.git
HEAD: d0a0bbef8db45cca303c607037338d02d8eaa8ca
Last commit: 7 weeks ago
```

This is your problem. You are using your own fork of `brew`.

macOS 13 support was added in 3.5.1.

**Reply by leavelet** [reply]
got it.
I totally forgot it, thank you very much!
